### PR TITLE
test: Document EVM unit test fixture

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by halfalicious** [comment]
cc @chfast

File: test/unittests/evm_fixture.hpp [hunk 1]
```diff
@@ -34,15 +34,33 @@
...
     evm() noexcept : vm{get_vm()} {}
 
-    /// Wrapper for evmone::execute. The result will be in the .result field.
+
+    /// Executes given code.
+    ///
+    /// @param gas        The gas limit for execution.
+    /// @param code       The EVM bytecode.
+    /// @param input_hex  The hex encoded EVM input.
+    /// The execution results is kept in .result filed. The .gas_used is updated accordingly.
```
Comment (axic) [inline test/unittests/evm_fixture.hpp]: Typo: filed -> field

**Comment by chfast** [comment]
> Why not `uint64_t `for gas (since I don't think we ever have negative gas)?

This is how EVMC defines it. The signed type is better for arithmetic (https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es102-use-signed-types-for-arithmetic) and especially for gas counting checking if gas went negative is super useful.

File: test/unittests/evm_fixture.hpp [hunk 2]
```diff
@@ -34,15 +34,33 @@
...
 
     evm() noexcept : vm{get_vm()} {}
 
-    /// Wrapper for evmone::execute. The result will be in the .result field.
+
+    /// Executes given code.
+    ///
+    /// @param gas        The gas limit for execution.
+    /// @param code       The EVM bytecode.
+    /// @param input_hex  The hex encoded EVM input.
```
[the remark this answers is not in the document]
Comment (axic) [reply test/unittests/evm_fixture.hpp]: Perhaps just say it is the calldata?

File: test/unittests/evm_fixture.hpp [hunk 3]
```diff
@@ -34,15 +34,33 @@
...
+    /// The execution results is kept in .result filed. The .gas_used is updated accordingly.
     void execute(int64_t gas, bytes_view code, std::string_view input_hex = {}) noexcept
     {
         auto input = from_hex(input_hex);
```
[the remark this answers is not in the document]
Comment (axic) [reply test/unittests/evm_fixture.hpp]: I think perhaps `.msg` superseded it. @chfast ?

**Comment by chfast** [comment]
Continuation in https://github.com/ethereum/evmone/pull/170.

**Pull request merged**